docker-micro-benchmark/helpers/docker_helpers.go

```diff
@@ -17,40 +17,75 @@ limitations under the License.
 package helpers
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
 	"math/rand"
 	"strconv"
 	"sync"
 	"time"
 
-	docker "github.com/fsouza/go-dockerclient"
+	docker "github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/container"
+	"github.com/docker/engine-api/types/events"
+	"github.com/docker/engine-api/types/network"
+	"github.com/docker/engine-api/types/strslice"
 	"github.com/juju/ratelimit"
+	"golang.org/x/net/context"
+)
+
+const (
+	TestImage   = "ubuntu"
+	TestCommand = "ubuntu"
 )
 
 func newContainerName() string {
 	return "benchmark_container_" + strconv.FormatInt(time.Now().UnixNano(), 10) + strconv.Itoa(rand.Int())
 }
 
+func getContext() context.Context {
+	return context.Background()
+}
+
+func PullTestImage(client *docker.Client) {
+	rc, err := client.ImagePull(getContext(), types.ImagePullOptions{
+		ImageID: TestImage,
+		Tag:     "latest",
+	}, nil)
+	if err != nil {
+		fmt.Errorf("%s", err)
+		return
+	}
+	defer rc.Close()
+	decoder := json.NewDecoder(rc)
+	var msg events.Message
+	for {
```

Comment:
Yeah, I saw that, and just tried `events.Message` here and found it partially works. :)
In fact I don't think there api is stabilized now, also see:
https://github.com/docker/engine-api/pull/140